components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -185,12 +185,14 @@ void ESP8266Interface::_connect_async()
     }
     _connect_retval = _esp.connect(ap_ssid, ap_pass);
     if (_connect_retval == NSAPI_ERROR_OK || _connect_retval == NSAPI_ERROR_AUTH_FAILURE
-            || _connect_retval == NSAPI_ERROR_NO_SSID) {
+            || _connect_retval == NSAPI_ERROR_NO_SSID || (_conn_timer.read_ms() >= ESP8266_CONNECT_TIMEOUT)) {
```

Comment:
Should check the _if_blocking  variable. In case the interface is configured as non-blocking giving up isn't an option.

Reply:
I wonder if we should perhaps provide some timeout possibility for the blocking calls? Right now connect() might just last forever and there is nothing user can do about it...

Reply:
_if_blocking is true by default, so our wifi tests in CI are back to failing when I check the _if_blocking flag before timing out...

Reply:
@VeijoPesonen , just to make sure - you meant that when interface is configured to **blocking**, then giving up isn't an option?

Reply:
When the call is non-blocking then giving up isn't an option as the system can continue with what ever it was doing. From my point of view timing out with the blocking call is fine - the timeout needs to be just big enough maybe something like 100s or so. @kjbracey-arm your opinion would be greatly appreciated here.

Reply:
@VeijoPesonen , I misinterpreted your idea and assumed a typo, sorry about that... Now I see your point.
The ESP8266_CONNECT_TIMEOUT was set to 15s so far, so I left it this way, assuming it has been discussed before. But we can make it anything that is more reasonable. If it's a large number however I will also need to adjust the wifi test timeout, I think.
@kjbracey-arm , waiting for your opinion.

Reply:
It would be useful that, if there is some significant timeouts, those would be documented somewhere.

Reply:
@teetak01 , thanks for your input. I will add documentation to the `connect()` doxygen description.
Also - I found that this ESP8266_CONNECT_TIMEOUT define is actually coming from the ESP8266 class and is heavily used there, so I will define a new one for the purpose of our higher-level connect timeout.